packages/core/src/utils/prepareEvent.ts

```diff
@@ -34,10 +35,7 @@ export function prepareEvent(
   };
 
   applyClientOptions(prepared, options);
-  applyIntegrationsMetadata(
-    prepared,
-    options.integrations.map(i => i.name),
-  );
+  applyIntegrationsMetadata(prepared, integrations);
```

Comment:
If we added integrations as optional field to `hint`, we could basically either use that or fall back to current behavior if it is unset?